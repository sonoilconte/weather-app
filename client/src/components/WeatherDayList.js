import React from 'react';
import WeatherDay from './WeatherDay';

class WeatherDayList extends React.Component {
	render() {
		return (
			<div className="row justify-content-center">
				<div className="col-md-6">
					<div className="status-message">{this.props.statusMessage}</div>
					{this.props.weatherDays.map((day, index) => <WeatherDay key={index} day={day} />)}
				</div>
			</div>
		);
	}
}

export default WeatherDayList;