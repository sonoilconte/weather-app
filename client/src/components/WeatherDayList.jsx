import React from 'react';
import PropTypes from 'prop-types';
import WeatherDay from './WeatherDay';

function WeatherDayList({ statusMessage, weatherDays }) {
    return (
        <div className="row justify-content-center">
            <div className="col-md-6">
                <div className="status-message">{statusMessage}</div>
                {
                    weatherDays.map(
                        (day, index) => (
                            <WeatherDay
                                // eslint-disable-next-line react/no-array-index-key
                                key={index}
                                date={day.date}
                                high={day.high}
                                low={day.low}
                            />
                        ),
                    )
                }
            </div>
        </div>
    );
}

WeatherDayList.propTypes = {
    statusMessage: PropTypes.string.isRequired,
    // eslint-disable-next-line react/forbid-prop-types
    weatherDays: PropTypes.array.isRequired,
};

export default WeatherDayList;
