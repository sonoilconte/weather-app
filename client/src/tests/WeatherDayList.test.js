import renderer from 'react-test-renderer';

import WeatherDayList from '../components/WeatherDayList';

it('renders the WeatherDayList correctly with weather data', () => {
	const statusMessage = 'Showing results for Denver, Colorado, United States';
	const data = [
		{ date: '2022-10-15', high: 60.1, low: 50.2 },
		{ date: '2022-10-16', high: 61.1, low: 51.2 },
		{ date: '2022-10-17', high: 62.1, low: 52.2 }
	];
	const tree = renderer.create(<WeatherDayList statusMessage={statusMessage} weatherDays={data} />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

it('renders the WeatherDayList correctly with an empty array for data', () => {
	const statusMessage = 'No results found';
	const tree = renderer.create(<WeatherDayList statusMessage={statusMessage} weatherDays={[]} />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});