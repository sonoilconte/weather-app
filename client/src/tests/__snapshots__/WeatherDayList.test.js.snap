// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the WeatherDayList correctly with an empty array for data 1`] = `
<div
  className="row justify-content-center"
>
  <div
    className="col-md-6"
  >
    <div
      className="status-message"
    >
      No results found
    </div>
  </div>
</div>
`;

exports[`renders the WeatherDayList correctly with weather data 1`] = `
<div
  className="row justify-content-center"
>
  <div
    className="col-md-6"
  >
    <div
      className="status-message"
    >
      Showing results for Denver, Colorado, United States
    </div>
    <div
      className="card"
    >
      <div
        className="card-body"
      >
        <div
          className="card-title"
        >
          Fri, October 14, 2022
        </div>
        <div
          className="card-text"
        >
          High: 
          60.1
          °F
        </div>
        <div
          className="card-text"
        >
          Low: 
          50.2
          °F
        </div>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="card-body"
      >
        <div
          className="card-title"
        >
          Sat, October 15, 2022
        </div>
        <div
          className="card-text"
        >
          High: 
          61.1
          °F
        </div>
        <div
          className="card-text"
        >
          Low: 
          51.2
          °F
        </div>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="card-body"
      >
        <div
          className="card-title"
        >
          Sun, October 16, 2022
        </div>
        <div
          className="card-text"
        >
          High: 
          62.1
          °F
        </div>
        <div
          className="card-text"
        >
          Low: 
          52.2
          °F
        </div>
      </div>
    </div>
  </div>
</div>
`;
